bpf/ipsec.h

```diff
@@ -0,0 +1,186 @@
+/*
+ * IPsec monitoring kretprobe eBPF hook.
+ */
+
+#ifndef __IPSEC_H__
+#define __IPSEC_H__
+
+#include "utils.h"
+
+static inline int ipsec_lookup_and_update_flow(flow_id *id, int flow_encrypted_ret,
+                                               u16 eth_protocol) {
+    additional_metrics *extra_metrics = bpf_map_lookup_elem(&additional_flow_metrics, id);
+    if (extra_metrics != NULL) {
+        extra_metrics->end_mono_time_ts = bpf_ktime_get_ns();
+        extra_metrics->eth_protocol = eth_protocol;
+        extra_metrics->flow_encrypted_ret = flow_encrypted_ret;
+        extra_metrics->flow_encrypted = flow_encrypted_ret == 0 ? true : false;
```

Comment:
if a flow contains both encrypted and unencrypted packets, here we keep only the last packet state.
I'd argue it's more important to show any unencrypted state, no?
So something like:
```
if (extra_metrics->flow_encrypted && flow_encrypted_ret != 0) {
extra_metrics->flow_encrypted = false;
}
```
wdyt?

Reply:
the encrypted flag name my be missleading here as it work both ways
for xfrm_input where we decrypt and check decryption state, while in xfrm_out this the encryption path , maybe I should rename them here as well to ipsec_ret and ipsec_success similar to the json ?